wurst/objects/items/PlagueBomb.wurst

```diff
@@ -0,0 +1,160 @@
+        ..setActivelyUsed(true)
+        ..setCooldownGroup("charged")
+        ..setHitPoints(75)
+        ..setLumberCost(100)
+        ..setGoldCost(0)
+        ..setLevelUnclassified(2)
+        ..setPriority(41)
+        ..setTooltipBasic("Trade for "+TOOLTIP)
+        ..setNumberofCharges(1)
+        ..setPerishable(true)
+        ..setCanBeDropped(true)
+        ..setIgnoreCooldown(true)
+        ..setCanBeSoldToMerchants(true)
+        ..setCanBeSoldByMerchants(true)
+
+//Create the projectile which drives the rest of the spell
+function createMissile(unit caster, vec3 targetPos)
+    vec3 initialPos = caster.getPos().withTerrainZ(MISSILE_FLY_HEIGHT)
+    real distance = targetPos.distanceTo2d(initialPos)
+    angle currentAngle = caster.getFacingAngle()
+
+    effect missileEffect = addEffect(MODEL , initialPos)
+    vec3 distanceVector = targetPos - initialPos
+
+    //Move the projectile to its target location.
+    doPeriodically(0.02) (CallbackPeriodic cb) ->
+        vec3 currentPos = missileEffect.getPos3()
+        currentPos += PROJECTILE_INTERVAL * distanceVector
+        //Add rotation to the projectile
+        currentAngle += 5..asAngleDegrees()
+
+        missileEffect.setOrientation(currentAngle, currentAngle, currentAngle)
+        missileEffect.setPos(currentPos)
+
+        //If the projectile is at target position.
+        if currentPos.distanceTo(targetPos) < 0.1
+            //Stop the loop.
+            destroy cb
+            //Destroy the effect.
+            missileEffect.destr()
+            //Create the plague cloud at target location.
+            createPlagueCloud(targetPos)
+            //Start the process of infecting meat/buildings.
+            replaceMeat(targetPos.toVec2(), caster)
+        
+
+//Creates and destroys the plague cloud effect at target location.
+function createPlagueCloud(vec3 targetPos)
+    effect plagueEffect = addEffect(EFFECT, targetPos)
+    plagueEffect.setScale(1)
+    doAfter(EFFECT_DURATION) -> 
+        plagueEffect.destr()
+
+//Finds and replaces all meat on the ground and in buildings.
+function replaceMeat(vec2 targetPos, unit caster)
+    var totalMeat = 0
+    let meatList = new LinkedList<item>
+    //Find all enemy buildings in target location.
+    forUnitsInRange(targetPos, ABILITY_RADIUS) (unit u) ->
+        if u.isStructure() and u.isEnemyOf(caster.getOwner())
+            //Add an effect to show which buildings were affected.
+            effect buildingCloud = u.addEffect(EFFECT, "origin")
+            doAfter(EFFECT_DURATION) -> 
+                buildingCloud.destr()
+                
+            
+            //Get the inventory for each building found.
+            let map = u.getInventory()
+            map.forEach() (item t) ->
+                if t.getTypeId() == ITEM_COOKED_MEAT
+                    meatList.add(t)
+    for meat in meatList
+        totalMeat += meat.getCharges()
+
+    let meatToConvert = (totalMeat / 10).floor()
```

Comment:
I think that this should be `ceiling` to make the effect stronger.

Reply:
If it was using ceiling having 11 meat in the building would cause 12 to get deleted

Reply:
The difference for 11 meat would be 1 versus 2. It just means that every stack is reduced instead of only full stacks.

Reply:
I'm confused by what you're saying.with 11 meat ceil would make it get reduced by 2 meat every 10 seconds, floor makes it get reduced by 1 every second.

Reply:
You mean "reduced by 1 ever 10 seconds" don't you?

Yes, the goal is to treat partial stacks like full stacks, so 11 meat over 2 stacks would be reduced by 2 total during a tick.

Reply:
I see that you were talking about the total before. I was assuming the decay would be adjusted per tick, e.g. 21 -> 18 -> 16 -> 14 -> 12 -> 10 over 5 ticks.